### ISSUE TITLE
Keyboard focus remains in same location after mouse click elsewhere in app

### ISSUE THREAD

**Issue by chiaramooney** [microsoft/react-native-windows#14497]
### Problem Description

When navigating through an RNW using the keyboard a black focus ring appears around the currently focused control. However, if the user clicks on a different component in the app, the focus ring remains visible on the previous control. This behavior does not match other Windows app like WinUI3 Gallery. 

Video Example: 

https://github.com/user-attachments/assets/d907ef94-656a-4d26-8b3f-e420bf598b64

WinUI3 Behavior: 

https://github.com/user-attachments/assets/c78df80f-521b-468d-8194-155bbd82f134

### Steps To Reproduce

1. Use keyboard to navigate through app, click on a new control.

### Expected Results

Focus ring should disappear from screen.

### CLI version

N/A

### Environment

```markdown
N/A
```

### Target Platform Version

None

### Visual Studio Version

None

### Build Configuration

None

**Comment by chrisglein** [comment]
Seems related to [#14496](https://github.com/microsoft/react-native-windows/issues/14496).

Mouse clicking doesn't appear to moving keyboard focus. Seems like an underlying issue.

This repro'd in main, so I'd expect that it has this change from @acoates-ms : https://github.com/microsoft/react-native-windows/pull/14359
Question is whether that didn't catch all cases, or there was a regression.
